packages/jupyter-chat/src/components/input/use-chat-commands.tsx

```diff
@@ -101,34 +100,20 @@ export function useChatCommands(
       return;
     }
 
-    const cursorIndex = inputRef.current?.selectionStart;
-    if (
-      !chatCommandRegistry ||
-      cursorIndex === null ||
-      cursorIndex === undefined
-    ) {
+    if (!chatCommandRegistry) {
       return;
     }
 
-    const currentWord = getCurrentWord(input, cursorIndex);
+    const currentWord = inputModel.currentWord;
     if (!currentWord) {
       return;
     }
 
-    if (command.replaceWith) {
-      replaceCurrentWord(input, cursorIndex, command.replaceWith, setInput);
-      return;
-    }
-
-    const replaceCurrentWordClosure = (newWord: string) => {
-      replaceCurrentWord(input, cursorIndex, newWord, setInput);
-    };
+    // When a command is selected, the input value is not updated to avoid conflict
+    // changes. It must be updated here, even if there is no replacement string.
+    inputModel.replaceCurrentWord(command.replaceWith ?? command.name);
```

Comment:
This change is to avoid concurrency on the input update.
When a command is selected from the list, the `onInputChange()` callback of the `ChatInput` is called first with the value (name) of the command. Then the `onChange()` callback of the `Autocomplete` is called. This resulted to a replacement of the whole input.
To avoid this behavior, the `onInputChange()` does not update the input value when change come from a selected command (using the `reason` props, set to `reset` in that case), and the value is updated in the `onchange()`, whether it need to be replaced or not.

This seems to work smoothly, but maybe it deserves some more testing to ensure it did not break anything else.

cc. @dlqqq

Reply:
@brichet Thank you for documenting this. I've tested it locally and it works well. The code highlighted here actually introduces a small regression, because we only want to replace the current word if `replaceWith` is set. If it's not set, then the handling of the command should be deferred to the command provider. I've pushed this change to your branch, and have verified this locally.

Note: I actually added that `if (reason === 'reset') { ... }` condition in the original PR, but removed it later because I found that it wasn't necessary. However, based on my local testing, this PR definitely needs that condition. I'm not sure why this bug only surfaced here in this PR. Leaving this as a note in case this breaks in the future.

Reply:
> because we only want to replace the current word if replaceWith is set.

Let's keep your change, I don't have a strong opinion on it :+1: